drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/project_unit_tracker.rb

```diff
@@ -53,13 +58,24 @@ def assign_next_unit(enrollment_pk:, unit_type_mper_id:, fallback_start_date: ni
       unit_pk
     end
 
+    def assign_next_unit_to_new_enrollment(enrollment:, project_pk:, unit_type_mper_id:)
```

Comment:
So the idea with this tracker/deferred-loader class is that it accumulates the occupancies and processes them all at the end, rather than creating them within the importer. 

The problem is that the deferred loader destroys all occupancy records before it runs, so the occupancies we are building here will be deleted later. This doesn't show in the test because the deferred loader only runs if assignments are present.

I have some ideas on how to fix this:
* modify the deferred DeferredProjectUnitOccupancyLoader so it only destroys occupancies that were assigned (this might leave some cruft from old records)
* rather than creating occupancies in the tracker/referral loader, let's defer occupancy creation to the end (we can do that since we are creating the enrollments one at a time)